From 81eaab0a39a398182c29407446699b37abb6efd2 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Sat, 23 Jan 2016 00:13:07 +0800
Subject: [PATCH 0079/1345] arm64: dts: add support for SDIO in Armada 37000
 DT files

commit  b8d67960411771b32d6cdef6e1e31291fd2054ea from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add south bridge SDIO node in DT files and enable it for DB board.
Card detection is done by SW polling "broken_cd" mode.

Change-Id: Iaba1a5b9edf27905ec80ab3d8ecc80713ea36c1f
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27684
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index a89855f..33d136e 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -161,6 +161,19 @@
 	status = "okay";
 };
 
+&sdhci1 {
+	wp-inverted;
+	bus-width = <4>;
+	cd-gpios = <&gpiosb 2 GPIO_ACTIVE_LOW>;
+	vqmmc-supply = <&vcc_sd_reg1>;
+	mmc-pwrseq = <&sd_pwrseq>;
+	marvell,pad-type = "sd";
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdio_pins>;
+};
+
 /* CON31 */
 &usb3 {
 	status = "okay";
-- 
1.7.9.5

